tests/unit/remote/test_s3.py

```diff
@@ -0,0 +1,102 @@
+import boto3
+import os
+import pytest
+import mock
+from moto import mock_s3
+
+from dvc.remote.s3 import RemoteS3
+
+
+@pytest.fixture
+def s3():
+    """Returns a connection to a bucket with the following file structure:
+
+        bucket
+        ├── data
+        │  ├── alice
+        │  ├── alpha
+        │  └── subdir
+        │     ├── 1
+        │     ├── 2
+        │     └── 3
+        ├── empty_dir
+        ├── empty_file
+        └── foo
+    """
+    # Mocked AWS Credentials for moto.
```

Comment:
@mroutis But you don't need these. You don't even need `with mock_s3()`, you can simply use `@mock_s3` decorator here, that way it is guaranteed that you are not using regular unpatched boto3 without a need to bloat tests with unnecessary code.

Reply:
@efiop , tried to use the decorator like this:

```python
@pytest.fixture
@mock_s3
def s3():
```

The decorator should be equivalent to `with mock_s3`.

I'll submit another commit without the environment variables, if you think those aren't needed

Reply:
@mroutis You are right, we have to use `with mock_s3()`.